### PR TITLE
add API to get node's route and ip

### PR DESCRIPTION
**Pull request by jenniexie**
I realize I did not add happy API for getting node ip and node route
In https://github.com/openweave/happy/pull/37, I only added command line support to get node ip and node route.
This PR will add Happy API to get node ip and route.

### CODE REVIEW

File: happy/HappyNodeRoute.py [hunk 1]
```diff
@@ -490,15 +490,21 @@ def __update_state(self):
         else:
             self.removeNodeRoute(self.node_id, self.to)
 
+    def getNodeRoute(self, node_id, route_type):
```
Comment (gerickson) [inline happy/HappyNodeRoute.py:493]: This needs API documentation.

File: happy/State.py [hunk 2]
```diff
@@ -181,6 +181,17 @@ def getNodeInterfaceAddresses(self, interface_id, node_id=None, state=None):
             return []
         return node_interface["ip"].keys()
 
+    def getNodeAddrMatchPrefix(self, node_id, interface, prefix):
```
Comment (gerickson) [inline happy/State.py]: This needs API documentation.

Comment (gerickson) [reply happy/State.py]: Suggest renaming this `getNodeAddrMatchingPrefix`.

**Review by gerickson — CHANGES_REQUESTED** [review]
A few API documentation and naming changes are requested.

File: happy/HappyNodeRoute.py (hunk 1 above)
Comment (jenniexie) [reply happy/HappyNodeRoute.py:493]: done

File: happy/State.py (hunk 2 above)
Comment (jenniexie) [reply happy/State.py]: done

File: happy/HappyNodeRoute.py [hunk 3]
```diff
@@ -490,15 +490,30 @@ def __update_state(self):
...
+        """This function will get node route base on node_id and route_type.
```
Comment (robszewczyk) [inline happy/HappyNodeRoute.py:494]: nit: typo `based` rather than `base`

**Review by robszewczyk — APPROVED** [review]

File: happy/HappyNodeRoute.py (hunk 3 above)
Comment (gerickson) [reply happy/HappyNodeRoute.py]: @jenniexie, thanks for taking a first pass at this. I don't know if there's a Python documentation standard; however, at minimum, I'd expect something Doxygen-like here that includes a @brief description, @param declarations describing each input and output parameter, and a @return/@retval describing the return value(s).

**Review by gerickson — APPROVED** [review]

**Comment by Vyrastas** [comment]
Not to delay the PR, but Python commenting for Doxygen is a little different than C, we should make sure it's done like this if we're aiming to generate a reference out of Happy:

http://www.doxygen.nl/manual/docblocks.html#pythonblocks

**Comment by jenniexie** [comment]
> Not to delay the PR, but Python commenting for Doxygen is a little different than C, we should make sure it's done like this if we're aiming to generate a reference out of Happy:
> 
> http://www.doxygen.nl/manual/docblocks.html#pythonblocks

Hi Jeff, Thanks for your comment, I used same format of python documentation which our Nest Engprod team use internally, also we have the same format in Happy code, example is IP.py.
I will follow up with you tomorrow and see what should be the standard python document for external like on github. We can consider it as future improvement and address it in a separate PR.

File: happy/HappyNodeRoute.py (hunk 3 above)
Comment (jenniexie) [reply happy/HappyNodeRoute.py:494]: fixed

**Review by gerickson — APPROVED** [review]

**Pull request merged**